packages/cli/src/webhooks/webhook-last-node-response-extractor.ts

```diff
@@ -0,0 +1,208 @@
+import { Container } from '@n8n/di';
+import get from 'lodash/get';
+import { BinaryDataService } from 'n8n-core';
+import type {
+	INode,
+	INodeExecutionData,
+	ITaskData,
+	IWebhookData,
+	IWorkflowDataProxyAdditionalKeys,
+	Result,
+	WebhookResponseData,
+	Workflow,
+	WorkflowExecuteMode,
+} from 'n8n-workflow';
+import { BINARY_ENCODING, createResultError, createResultOk, OperationalError } from 'n8n-workflow';
+import type { Readable } from 'node:stream';
+
+type NonStreamResponse = {
+	type: 'nonstream';
+	body: unknown;
+	contentType: string | undefined;
+};
+
+type StreamResponse = {
+	type: 'stream';
+	stream: Readable;
+	contentType: string | undefined;
+};
+
+/**
+ * Extracts the response for a webhook when the responde mode is set to
+ * `lastNode`.
+ */
+export async function extractWebhookLastNodeResponse(
+	responseDataType: WebhookResponseData | undefined,
+	lastNodeRunData: ITaskData,
+	workflow: Workflow,
+	workflowStartNode: INode,
+	webhookData: IWebhookData,
+	executionMode: WorkflowExecuteMode,
+	additionalKeys: IWorkflowDataProxyAdditionalKeys,
+): Promise<Result<NonStreamResponse | StreamResponse, OperationalError>> {
+	if (responseDataType === 'firstEntryJson') {
+		return extractFirstEntryJsonFromRunData(
+			lastNodeRunData,
+			workflow,
+			workflowStartNode,
+			webhookData,
+			executionMode,
+			additionalKeys,
+		);
+	}
+
+	if (responseDataType === 'firstEntryBinary') {
+		return await extractFirstEntryBinaryFromRunData(
+			lastNodeRunData,
+			workflow,
+			workflowStartNode,
+			webhookData,
+			executionMode,
+			additionalKeys,
+		);
+	}
+
+	if (responseDataType === 'noData') {
+		return createResultOk({
+			type: 'nonstream',
+			body: undefined,
+			contentType: undefined,
+		});
+	}
+
+	// Default to all entries JSON
+	return extractAllEntriesJsonFromRunData(lastNodeRunData);
+}
+
+/**
+ * Extracts the JSON data of the first item of the last node
+ */
+function extractFirstEntryJsonFromRunData(
+	lastNodeRunData: ITaskData,
+	workflow: Workflow,
+	workflowStartNode: INode,
+	webhookData: IWebhookData,
+	executionMode: WorkflowExecuteMode,
+	additionalKeys: IWorkflowDataProxyAdditionalKeys,
+): Result<NonStreamResponse, OperationalError> {
+	if (lastNodeRunData.data!.main[0]![0] === undefined) {
```

Comment:
These accesses with `!` are error prone. Is there a safer way to access the item?